apps/omg_watcher/lib/omg_watcher/coordinator_setup.ex

```diff
@@ -23,16 +23,11 @@ defmodule OMG.Watcher.CoordinatorSetup do
 
     %{
       depositor: [finality_margin: deposit_finality_margin],
-      convenience_deposit_processor: [waits_for: [:depositor], finality_margin: deposit_finality_margin],
       block_getter: [
-        waits_for: [depositor: :no_margin, convenience_deposit_processor: :no_margin],
+        waits_for: [depositor: :no_margin],
         finality_margin: 0
       ],
       exit_processor: [waits_for: :depositor, finality_margin: finality_margin],
```

Comment:
do we have a mechanism that ensures that exits from a given eth height are processed after blocks submitted at that eth height?

I think not, `ExitStarted` events will go straight to the `Consumer` without waiting. (the `exit_processor` listener :point_up:, which will trigger that, doesn't wait for blocks **by design**, so one cannot just add a `waits_for: block_getter` there :( ).

We either have to ensure that the order those events are consumed doesn't matter (it used to matter due to some key constraints not being met otherwise) or force a specific order of consumption somehow.

Reply:
@pdobacz I'm not sure, is this an issue caused by the split? It looks like it's been behaving this way prior.

If so I'll create a new issue to address this comment.

Reply:
also to expand @unnawut why was this needed @pdobacz 
```
convenience_exit_processor: [
        waits_for: [:convenience_deposit_processor, :block_getter],
        finality_margin: finality_margin
      ],
```

Reply:
I'm afraid it is a byproduct of the split, and the way the ethereum events are written to the `Bus`. If I'm not mistaken, this kind of happens "by-the-way" - when a listener, bound to a particular eth event, does its job, it will push an event into the bus (`publish...`).

Before the split, there were 2 listeners that were listening to `ExitStarted` - `exit_processor` and `convenience_...`. `exit_processor` wants those events without and `convenience_...` with waiting on the `block_getter` service. This can be read from this big coordinator setup map here, prior to the change.

Now what happens when you split and remove `convenience_...`? The `ExitStarted` events are getting written to the `Bus` when `exit_processor` listener asks for them - so without waiting on `block_getter`.

Hope this makes sense, and if I missed something, LMK.

Actually, there is a broader issue to related to the `this kind of happens "by-the-way" - when a listener, bound to a particular eth event, does its job, it will push an event into the bus` part:

There is no requirement that all eth events will have their listeners, nor that they will have only one. So if there's >1 listener for a particular eth event (like when we want to process an event with 2 different waiting/finality semantics, which was the case of `ExitStarted`), that event will get pushed to the `Bus` twice.

Hmmm, what to do about this? What do you think are the expectations of the `WatcherInfo` (or other consumers of the `Bus`)?

Maybe, the `Bus` should be fed events by a dedicated `EthereumEventListener` that grabs _all_ the events from the contract(s). We can then give it some appropriate waiting semantics :thinking:

Reply:
Okay Ino just pointed out to me that the `:convenience_exit_processor` below does wait for block getter.

But now that the Info API is eventually consistent, and should not be used for security-critical purposes, this might not be an issue?

Reply:
☝️ Ugh my comment above was written before @pdobacz reply

Reply:
> Maybe, the Bus should be fed events by a dedicated EthereumEventListener that grabs all the events from the contract(s). We can then give it some appropriate waiting semantics 🤔

I still think event ordering should not be imposed/expected by the bus, in order to maintain its asynchronicity and it should be up to the bus consumer (i.e. Info API) how it wants to handle the events.

I'm thinking of 2 options here:

1. Let it be. The Info API is eventually consistent and not used for security-critical purposes, so delay between blocks and exits showing up is acceptable.

2. Ensure Info API processes the block before processing their exits. Implement a `wait_for:` kind of mechanism between `BlockConsumer` and `ExitConsumer`.

I prefer 1 for its simplicity though.

Reply:
OK, lets continue...
1. why exactly did `convenience_exit_processor` need to wait for `block_getter`? and what was the general idea behind `waiting semantic`?
2. Is this somehow related to ethereum reorgs?
3. What is the consequence if *exactly the same* event gets published twice by two different ethereum_event_listeners, apart from ensuring they're not inserted twice on _watcher_info_?

Reply:
1. I think this was required by the way exits were inserted into the DB. It may be a leftover though, come to think of it. So if you can insert into `WatcherInfo`'s DB first an exit, and then a tx creating the corresponding UTXO, then all is good. That would be `ensure that the order those events are consumed doesn't matter` I wrote above and @unnawut 's point (1.).
2. nope, reorgs aren't involved. If reorgs deeper than `finality_margin` happen heavily impact the events (like, drop an exit), it is problematic anyway
3. astonishment, I suppose. This would require the consumer to work with a less predictable service. This also couples the producer with the consumer (e.g. we for any reason stop handling some events in security - the producer - and suddenly the consumer would not get those events)

As much as locally, the problem would be solved by what I wrote in my (1.) above, I still would consider doing something like

> Maybe, the Bus should be fed events by a dedicated EthereumEventListener that grabs all the events from the contract(s)

This would provide the consumers with a very well described and predictable behavior - "all events are pushed exactly once at some defined point in time". Then consumers deal with that on their own.

Reply:
>    Maybe, the Bus should be fed events by a dedicated EthereumEventListener that grabs all the events from the contract(s)

>    This would provide the consumers with a very well described and predictable behavior - "all events are pushed exactly once at some defined point in time". Then consumers deal with that on their own.

Good points!

Reply:
I do think this is good for now and whatever design changes we do because of this, they can come in after this PR. Is there anything else @pdobacz or is this PR good to go from your perspective?

Reply:
ah yes and this. I think good for now as long as `ensure that the order those events are consumed doesn't matter (anymore)`. 

Separating a listener to cater for the `Bus` is totally a separate ticket

Reply:
Continue as https://github.com/omisego/elixir-omg/issues/1188